plugins/tls.js

```diff
@@ -57,21 +57,21 @@ exports.advertise_starttls = function (next, connection) {
 
         if (!dbr) return enable_tls();
 
-        // last TLS attempt failed
-        redis.del(dbkey); // retry TLS next connection.
-
         connection.results.add(plugin, { msg: 'no_tls'});
-        return next();
+        return next(CONT, 'STARTTLS disabled because previous attempt to set it up failed')
     });
 }
 
-exports.set_notls = ip => {
+exports.set_notls = function (connection) {
 
-    if (!tls_socket.cfg.redis) return;
```

Comment:
Leave this line here, else Haraka crashes while attempting to access a config property that doesn't exist in older config files / versions of Haraka.

Reply:
No, it won't, because that setting is specified in config boolean list https://github.com/haraka/Haraka/blob/master/tls_socket.js#L221

Reply:
👍